### PR TITLE
VDM-40: Send server start requests to the correct server.

### PR DESCRIPTION
**Pull request by manjujames**

### CODE REVIEW

File: lib/python/vdm/server/HTTPListener.py [hunk 1]
```diff
@@ -1889,6 +1914,7 @@ def main(runner, amodule, config_dir, server):
     SERVER_VIEW = ServerAPI.as_view('server_api')
     DATABASE_VIEW = DatabaseAPI.as_view('database_api')
     START_DATABASE_SERVER_VIEW = StartServerAPI.as_view('start_server_api')
+    START_LOCAL_SERVER_VIEW = StartLocalServerAPI.as_view('start_local_server_api')
```
Comment (akhanzode) [inline lib/python/vdm/server/HTTPListener.py]: Don't you need similar for recover? How does start local know its "start" or "recover"

File: lib/python/vdm/server/HTTPListener.py [hunk 2]
```diff
@@ -1633,8 +1633,13 @@ def put(database_id):
             Status string indicating if the database start request was sent successfully
         """
 
-        database = voltdbserver.VoltDBServer(database_id)
-        return database.start_database()
+        try:
+            database = voltdbserver.VoltDatabase(database_id)
+            return database.start_database(True)
```
Comment (akhanzode) [inline lib/python/vdm/server/HTTPListener.py]: Whats the purpose of flag?

File: lib/python/vdm/server/voltdbserver.py [hunk 3]
```diff
@@ -128,6 +128,24 @@ def start_database(self, recover=False):
         else:
             return create_response('Start request sent successfully to servers: ' + str(server_status), 200)
 
+    def start_server(server_id, recover=False):
```
Comment (akhanzode) [inline lib/python/vdm/server/voltdbserver.py]: Why not be explicit in action to use?

File: lib/python/vdm/server/voltdbserver.py [hunk 4]
```diff
@@ -128,6 +128,24 @@ def start_database(self, recover=False):
...
+        """
+        Sends start request to the specified server
+        """
+        action = 'start'
+        if recover:
+            action = 'recover'
+
+        try:
+            server = find_server(server_id)
```
Comment (akhanzode) [inline lib/python/vdm/server/voltdbserver.py]: what if find_server fails?

**Comment by akhanzode** [comment]
Rest looks alright.

File: lib/python/vdm/server/voltdbserver.py (hunk 3 above)
Comment (manjujames) [inline lib/python/vdm/server/voltdbserver.py]: This is called with false for startup and true for recover. Defaulting in the method declaration is just a convenience..

File: lib/python/vdm/server/HTTPListener.py (hunk 2 above)
Comment (manjujames) [inline lib/python/vdm/server/HTTPListener.py]: The flag says whether it is start in recover mode or not.

File: lib/python/vdm/server/HTTPListener.py (hunk 1 above)
Comment (manjujames) [inline lib/python/vdm/server/HTTPListener.py]: There is no public endpoint to recover a *server*. When we get request to recover a database, we call the internal endpoint directly.

**Comment by manjujames** [comment]
squash merged.

**Pull request closed without merging**